packages/cli/src/modules/insights/insights.module.ts

```diff
@@ -25,12 +30,9 @@ export class InsightsModule implements BaseN8nModule {
 		}
 	}
 
-	registerLifecycleHooks(hooks: ExecutionLifecycleHooks) {
-		const insightsService = this.insightsService;
-
-		hooks.addHandler('workflowExecuteAfter', async function (fullRunData) {
-			await insightsService.workflowExecuteAfterHandler(this, fullRunData);
-		});
+	@OnLifecycleEvent('workflowExecuteAfter')
+	async handleWorkflowExecuteAfter(ctx: WorkflowExecuteAfterContext) {
+		await this.insightsService.handleWorkflowExecuteAfter(ctx);
```

Comment:
we could add this decorator directly on `InsightsCollectionService.handleWorkflowExecuteAfter`, and remove the need to pass this context through two additional layers.